spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientConfiguration.java

```diff
@@ -155,6 +166,28 @@ public ServerIntrospector serverIntrospector() {
 		return new DefaultServerIntrospector();
 	}
 
+	@PostConstruct
+	public void preprocess() {
+		setProp(name, DeploymentContextBasedVipAddresses.key(), name);		
```

Comment:
And now I re-read your comments :-) Now I wonder why it's needed in both places.